### ISSUE TITLE
when canceling a dialog, warning displayed doesn't need a "X" close icon

### ISSUE THREAD

**Issue by lweitzman** [vmware/harbor#1795]
After testing the invalid input on a project creation, i tried to click on the cancel button but then got a warning that some fields were modified. in this case in might not be necessary. also the warning doesn't need an "X" in the right side. Also, the cancel button if clicked again should work.
@tonycgil can comment as well.

![screen shot 2017-03-24 at 10 56 12 am](https://cloud.githubusercontent.com/assets/626032/24299873/f3062f12-1080-11e7-8688-3cee9d7f8900.png)

**Comment by steven-zou** [comment]
@lweitzman 

  We can remove the 'X' icon button. After removing, do we need to add another choice 'NO' for user to keep on stay on the modal and dismiss the alert?

**Comment by lweitzman** [comment]
No, when the user clicks cancel they can start editing in the dialog again and the warning should go away. also, the user should be able to click the cancel button again to close the dialog

**Comment by steven-zou** [comment]
Still not get all the things. Let me try to clarify the scenario:

  User edit some fields, and click the 'cancel' for some reason;
  The inline alert appear to let user confirm if they really want to abandon the changes and close the dialog;
  If user click 'YES' in the inline alert, that means user confirms discard the changes;
  If user does not want to discard the changes and want to keep on staying on modal and editing contents,
   Questions coming, when to dismiss the alert?
   Click 'cancel' again to instead the 'YES' option?

**Comment by lweitzman** [comment]
@steven-zou i think in the modal dialog case, if the user clicks on cancel, you close the dialog. no need for a warning. but the issue is important when editing inline on the configurations pages. there when the user tries to navigate away from the changes, we need a warning dialog. no need for a button in the warning in that case.

**Comment by tonycgil** [comment]
I played with the interaction. Here’s my take:
1) If nothing has been inputed, then CANCEL should simply dismiss the dialog.
2) If something HAS been inputed and user clicks CANCEL for whatever reason, a warning should confirm if you want to dismiss and close what has been inputed. This will insure possible accidental CANCEL clicks from a user or allow for second thoughts of the action.
3) If you hit CANCEL again while the warning message is still there, it should blink to communicate to the user that he/she needs to dismiss the question first. 

Simply put, If I enter information on anything and then decide to CANCEL, I like to be asked if I’m sure I want to. I’ve had incidents where I cancel by accident and I loose the information I inputed.  
![warning b4 canceling](https://cloud.githubusercontent.com/assets/25991813/24364861/f6e04dee-12c8-11e7-99c0-a8b4adf5d823.png)

**Comment by steven-zou** [comment]
@tonycgil 

Remove close icon and add option 'NO'.
I don't think blink is necessary because the alert appeared is obvious for user to understand what happened now.

**Comment by lweitzman** [comment]
Can we use tony's simpler NO YES button links?

**Comment by steven-zou** [comment]
Sure, changed!
